projects/dotcms-field-elements/src/utils/props/validators/date.ts

```diff
@@ -0,0 +1,70 @@
+import { DotDateSlot } from '../../../models';
+
+const DATE_REGEX = new RegExp('\\d\\d\\d\\d-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])');
+const TIME_REGEX = new RegExp('^(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])$');
+
+/**
+ * Check if date is valid, returns a valid date string, otherwise null.
+ *
+ * @param string date
+ * @returns string
+ */
+export function dotValidateDate(date: string): string {
+    return DATE_REGEX.test(date) ? date : null;
+}
+
+/**
+ * Check if time is valid, returns a valid time string, otherwise null.
+ *
+ * @param string time
+ * @returns string
+ */
+export function dotValidateTime(time: string): string {
+    return TIME_REGEX.test(time) ? time : null;
+}
+
+/**
+ * Parse a data-time string that can contains 'date time' | date | time.
+ *
+ * @param string data
+ * @returns DotDateSlot
+ */
+export function dotParseDate(data: string): DotDateSlot {
+    const [dateOrTime, time] = data.split(' ');
```

Comment:
This split by just space could lead to issues, if we pass the string but separated for like 2 spaces instead of 1 will leave the value in a weird state.